src/AWS.Deploy.Common/Recipes/Validation/RecipeValidators/ECSFargateExistingResourcesValidator.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Linq;
+using System.Threading.Tasks;
+using AWS.Deploy.Common.Data;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Checks for existing resources that might conflict with the ECS Fargate deployment.
+    /// </summary>
+    public class ECSFargateExistingResourcesValidator : IRecipeValidator
+    {
+        public string ECSClusterName { get; set; } = "ECSCluster.NewClusterName";
```

Comment:
We should not be hard coding the id of the settings. The id used should be configured in the recipe. Something like the this.

```
    "Validators": [
        {
            "ValidatorType": "ElasticBeanstalkExistingResources",
	    "Parameters" :{
                  "ClusterName" : "ECSCluster.NewClusterName"
	     }
        }
    ],
```

This should be done for all recipe validators.

Reply:
Fixed. The recipe validator is now a generic one. It looks for option settings that have Resource validators and executes them. This is better than specifying individual setting IDs.